api4/channel_test.go

```diff
@@ -1485,39 +1485,40 @@ func TestUpdateChannelPrivacy(t *testing.T) {
 			CheckForbiddenStatus(t, resp)
 		})
 	}
+	th.TestForAllClients(t, func(t *testing.T, client *model.Client4) {
+		th.LoginTeamAdmin()
 
-	th.LoginTeamAdmin()
-
-	tt = testTable{
-		{"Converting default channel to private should fail", defaultChannel, model.CHANNEL_PRIVATE},
-		{"Updating privacy to an invalid setting should fail", publicChannel, "invalid"},
-	}
+		tt := testTable{
+			{"Converting default channel to private should fail", defaultChannel, model.CHANNEL_PRIVATE},
+			{"Updating privacy to an invalid setting should fail", publicChannel, "invalid"},
+		}
 
-	for _, tc := range tt {
-		t.Run(tc.name, func(t *testing.T) {
-			_, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
-			CheckBadRequestStatus(t, resp)
-		})
-	}
+		for _, tc := range tt {
+			t.Run(tc.name, func(t *testing.T) {
+				_, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```

Comment:
The client should be replaced with the variable injected by the helper:

```suggestion
				_, resp := client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```